includes/User.php

```diff
@@ -2174,7 +2174,9 @@ public function setPassword( $str ) {
 		}
 
 		$this->setInternalPassword( $str );
-
+		if($clearTokens) {
+			wfRunHooks('SetPassword', [$this->getId()]);
```

Comment:
`SetPassword` hook wasn't used before. I added it only in setPassword method and disable when we don't want to invalidate tokens in Helios (creating account). I think that it should be fine.